core/src/main/java/org/testcontainers/utility/ResourceReaper.java

```diff
@@ -67,6 +68,8 @@ public void registerContainerForCleanup(String containerId, String imageName) {
      */
     public void stopAndRemoveContainer(String containerId) {
         stopContainer(containerId, registeredContainers.get(containerId));
+
+        registeredContainers.remove(containerId);
```

Comment:
Actually even though it's a little brittle I'd prefer to keep it that way. If `stopContainer` fails for some reason, my feeling is that it's slightly better to _potentially_ try again by leaving the containerId in the registered set.